sharktank/sharktank/layers/base.py

```diff
@@ -33,12 +41,82 @@ def _set_recursively_submodules_default_trace_tensor_key_prefix(
         )
 
 
-class BaseLayer(nn.Module):
+def get_model_type_id(model_type: type["BaseLayer"]) -> str:
+    """Get a string representation of the model type."""
+    return f"{model_type.__module__}.{model_type.__name__}"
+
+
+def create_model(config: PathLike | ModelConfig | Mapping[str, Any], /) -> "BaseLayer":
```

Comment:
Is `sharktank/layers/base.py` the right spot for these new `create_model` and `model_registry` symbols? 

Alternatives to consider:

* https://github.com/nod-ai/shark-ai/tree/main/sharktank/sharktank/models
* https://github.com/nod-ai/shark-ai/tree/main/sharktank/sharktank/tools
* https://github.com/nod-ai/shark-ai/tree/main/sharktank/sharktank/utils

I would expect a hierarchy of

1. Ops
2. Kernels
3. Layers
4. Models
5. Pipelines

Where this tooling would fit better under "Models" or "Pipelines".

Reply:
I agree that this seems like it'd fit better somewhere under the models directory. 

Tangent: ops and kernels are actually adjacent in the hierarchy, and that probably means we should rename that directory/do some reorganizing.

Reply:
I think this would make the most sense for a hierarchy:

1. ops
2. kernels
3. modules
    1. layers
    2. models
    3. (hypothetical) compiled pipelines -- in the event we do something like CompiledSDXLPipeline where "models" are assembled by invoking their entrypoints from within a MLIR "pipeline" -- will likely be more relevant with deployment outside of Python
4. pipelines

I think the refactor into "module" base-classing should be a separate (follow-up) PR.